services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/service/impl/PathManagerService.java

```diff
@@ -18,18 +18,122 @@
 
 package com.futurewei.alcor.dataplane.service.impl;
 
+import com.futurewei.alcor.dataplane.cache.SubnetPortsCacheV2;
+import com.futurewei.alcor.dataplane.cache.VpcPathCache;
+import com.futurewei.alcor.dataplane.cache.VpcSubnetsCache;
+import com.futurewei.alcor.dataplane.exception.InvalidPathModeException;
 import com.futurewei.alcor.web.entity.port.PortEntity;
+import com.futurewei.alcor.web.entity.subnet.InternalSubnetPorts;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
 import org.springframework.stereotype.Service;
 
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+
 @Service
+@Configuration
 public class PathManagerService {
+    @Autowired
+    private VpcSubnetsCache vpcSubnetsCache;
+
+    @Autowired
+    private SubnetPortsCacheV2 subnetPortsCache;
+
+    @Autowired
+    private VpcPathCache vpcPathCache;
+
     private static boolean USE_GRPC = true;
 
+    @Value("${path.mode}")
+    private String PATH_MODE;
+
+    @Value("${path.UPPER_VPC_SIZE}")
+    private int UPPER_VPC_SIZE;
+    @Value("${path.LOWER_VPC_SIZE}")
+    private int LOWER_VPC_SIZE;
+
     public boolean isFastPath() {
         return USE_GRPC;
     }
 
-    public boolean isFastPath(PortEntity portEntity) {
-        return USE_GRPC;
+    public boolean isFastPath(PortEntity portEntity) throws Exception{
+        switch (PATH_MODE) {
+            case "GRPC":
+                return USE_GRPC;
+            case "MQ":
+                return !USE_GRPC;
+            case "MIXED":
+                return choosePath(portEntity);
```

Comment:
@VanderChen The "Mixed" mode for me seems not a real MIXED mode, doesn't it? I think "AUTO" is better to describe this mode.